src/lib/y2partitioner/actions/controllers/btrfs_subvolume.rb

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2partitioner/actions/controllers/base"
+
+module Y2Partitioner
+  module Actions
+    module Controllers
+      # Controller class to deal with all stuff related to adding or removing devices to
+      # a Btrfs filesystem
+      class BtrfsSubvolume < Base
+        # @return [Y2Storage::Filesystems::Btrfs]
+        attr_reader :filesystem
+
+        attr_reader :subvolume
+
+        attr_accessor :subvolume_path
+
+        attr_accessor :subvolume_nocow
```

Comment:
Please add docs with the correct types, they are important for complex code like this

Reply:
This is still a WIP. Please, do not review it yet.